File: apps/api/src/tags/entities/tag.entity.ts
import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';

@Entity({ name: 'tags' })
export class Tag {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ type: 'text', unique: true, nullable: false })
  name: string;

  @Column({ type: 'text', nullable: true })
  description: string;
}

File: apps/api_go/internal/domain/tag_entity.go
package domain

import "gorm.io/gorm"

type Tag struct {
	gorm.Model
	Name        string  `gorm:"column:name;type:text;unique;not null"`
	Description *string `gorm:"column:description;type:text"`
}

func (Tag) TableName() string {
	return "tags"
}
